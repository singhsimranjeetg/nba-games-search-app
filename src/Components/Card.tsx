import styled from "styled-components";

const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
  margin: 1rem 0;
  border: solid 1px var(--surface);
  transition: box-shadow 300ms ease;
  border-radius: 0.5rem;
  height: min-content;
  background-color: var(--white);

  & label:not(.detail) {
    font-weight: 600;
  }

  @media (max-width: 768px) {
    width: 100%;
  }

  @media (min-width: 769px) {
    &:hover {
      box-shadow: var(--hover);
    }
  }
`;

interface CardProps {
  children?: React.ReactFragment | HTMLCollection | string;
  style?: React.CSSProperties | undefined;
}

const Card = ({ children, style }: CardProps) => (
  <CardContainer style={style}>{children}</CardContainer>
);

export default Card;
