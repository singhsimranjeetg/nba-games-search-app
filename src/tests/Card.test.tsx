import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Components/Card';

it('Snapshot tests Card', () => {
  expect(shallow(<Card />)).toMatchSnapshot();
});
