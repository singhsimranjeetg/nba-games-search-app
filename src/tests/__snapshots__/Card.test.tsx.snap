// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Card 1`] = `
<div>
  <styled.div />
</div>
`;
